### PR TITLE
[AEA-1334] fixes and features for MAM: auto remove project, project load check.

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fixes and features for MAM: auto remove project, project load check.

## Fixes

https://github.com/fetchai/agents-aea/issues/2296
https://github.com/fetchai/agents-aea/issues/2295

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/manager/manager.py [hunk 1]
```diff
@@ -249,20 +272,23 @@ def start_manager(
 
         :param local: whether or not to fetch from local registry.
         :param remote: whether or not to fetch from remote registry.
+
         :return: the MultiAgentManager instance.
         """
         if self._is_running:
-            return self
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:255]: That breaks compatibility. Why are we doing this? We could store this info in a property so user can request it.

File: aea/manager/project.py [hunk 2]
```diff
@@ -126,6 +129,22 @@ def builder(self) -> AEABuilder:
...
+            default_ledger = builder.get_default_ledger()
+            create_private_key(default_ledger, key_file)
+            if not builder.private_key_paths:
+                # no keys, but create one for builder only
+                builder.add_private_key(default_ledger, key_file)
+            if not builder.connection_private_key_paths:
+                builder.add_private_key(
+                    default_ledger, key_file, is_connection=True,
+                )
+            builder.build()
```
Comment (DavidMinarsch) [inline aea/manager/project.py:146]: I think we should just do a version check. Not sure why we do a whole build..

Comment (solarw) [reply aea/manager/project.py:146]: the whole build is quite better, it checks many things, even python code load

Comment (DavidMinarsch) [reply aea/manager/project.py:146]: How about just this:
```
self.install_pypi_dependencies()
self.build()
```

File: aea/manager/project.py [hunk 3]
```diff
@@ -126,6 +129,22 @@ def builder(self) -> AEABuilder:
         """Get builder instance."""
         return self._get_builder(self._get_agent_config(self.path), self.path)
 
+    def check(self) -> None:
+        """Check project is ok with builder.build."""
```
Comment (DavidMinarsch) [inline aea/manager/project.py:133]:
```suggestion
    def check(self) -> None:
        """Check we can still construct an AEA from the project with builder.build."""
```

File: aea/manager/manager.py [hunk 4]
```diff
@@ -352,6 +379,14 @@ def add_project(
...
+            project.check()
+        except Exception as e:
+            project.remove()
+            raise ProjectCheckError(
+                f"Failed to load project: {public_id} Error: {str(e)}", e
+            )
+
         if not restore:
             project.install_pypi_dependencies()
             project.build()
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:392]: change order

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**